### PR TITLE
Test Druid handling of decimal type with negative scale

### PR DESCRIPTION
**Pull request by kkauy**
## Summary

Add a reproducer test for negative-scale decimal handling in the Druid connector.

The test documents the current behavior: values ingested as `decimal(p, -s)` are read back through the connector as `VARCHAR`, so the decimal mapping branch is not reached.

## Related issue

Related to #28415

### CODE REVIEW

**Comment by kkauy** [comment]
Related to #28415

Adds an integration test covering negative-scale decimal handling in the Druid connector.
The test documents the current behavior: values ingested as `decimal(p, -s)` are read back through the connector as `VARCHAR`, so the decimal mapping branch is not reached.

**Comment by kkauy** [comment]
Hi, I submitted the signed CLA to cla@trino.io about two weeks ago. Could someone please check whether it was received and matched to my GitHub account (`kkauy`)? Thanks.

**Comment by kkauy** [comment]
This seems different from #28414.

In this Druid path, the integration test shows the columns come back as `jdbcType=12` (`VARCHAR`), so the DECIMAL/NUMERIC mapping branch is not reached.

This PR adds a reproducer test for #28415. I can adjust the test approach if a mergeable documentation-style version is preferred.

**Comment by Praveen2112** [comment]
But does Druid supports a dedicated decimal type OOB ? In the documentation it was mentioned that decimal is mapped to double.

**Comment by kkauy** [comment]
Good point. Based on the Druid SQL type documentation, `DECIMAL` maps to `DOUBLE` rather than fixed-point decimal semantics.

I will investigate  why this tested path comes back as `jdbcType=12` (`VARCHAR`) instead of `DOUBLE`.  Thanks in advance.

**Comment by findepi** [comment]
> In this Druid path, the integration test shows the columns come back as `jdbcType=12` (`VARCHAR`), so the DECIMAL/NUMERIC mapping branch is not reached.

thanks for checking this!
it might be it's just copy-pasted code. can we remove it then?

**Comment by kkauy** [comment]
That seems reasonable to me.

I checked and this pattern exists in a number of other JDBC connectors as well, so it does look copy-pasted. Based on the current Druid investigation, this branch appears unreachable here.

File: plugin/trino-druid/src/test/java/io/trino/plugin/druid/TestDruidTypeMapping.java [hunk 1]
```diff
@@ -184,6 +185,17 @@ public void testVarchar()
                 .execute(getQueryRunner(), druidCreateAndInsert("test_unbounded_varchar"));
     }
 
+    @Test
+    public void testDecimalNegativeScale()
```
Comment (ebyhr) [inline plugin/trino-druid/src/test/java/io/trino/plugin/druid/TestDruidTypeMapping.java]: Could you add roundtrip with positive scale, and rename this test to `testDecimal`? 

Also, please move this method above testVarchar.

Comment (ebyhr) [inline plugin/trino-druid/src/test/java/io/trino/plugin/druid/TestDruidTypeMapping.java:189]: It seems Druid maps unsupported types to varchar. The decimal type doesn't exist in (Druid runtime type) https://druid.apache.org/docs/latest/querying/sql-data-types/. 

I tested with random type name (e.g. `xxx`) and the test maps to varchar type instead of failure.

Comment (kkauy) [reply plugin/trino-druid/src/test/java/io/trino/plugin/druid/TestDruidTypeMapping.java:189]: Thanks for the clarification.

Since Druid does not expose decimal as a runtime type and unsupported types are mapped to VARCHAR, this test does not validate meaningful behavior.

I'll go ahead and close this PR.

**Comment by findepi** [comment]
We should remove the dead code:
- https://github.com/trinodb/trino/pull/28754
